File: Sem1/ProjetoIntegrador/exercicios/src/tarefaI/estudoDirigidoAlgoritimo/pg25_26Exercicio7/L01E.java
// e)Efetuar o cálculo e a apresentação do valor de uma prestação em atraso,
// utilizando a fórmula PRESTACAO ← VALOR + (VALOR * TAXA/100) * TEMPO).

package tarefaI.estudoDirigidoAlgoritimo.pg25_26Exercicio7;

import javax.swing.*;

public class L01E {
    static void main() {
    double installmentAmount;
    double interest;
    double valueTotal;
    int days;

     installmentAmount = Double.parseDouble(JOptionPane.showInputDialog("Digite o valor da prestação: ").replace(",","."));
     interest = Double.parseDouble(JOptionPane.showInputDialog("Digite taxa de juros: ").replace(",","."));
     days = Integer.parseInt(JOptionPane.showInputDialog("Digite quantos dias de atraso: "));

     valueTotal = simpleInterest(interest,installmentAmount,days)+installmentAmount;

     JOptionPane.showMessageDialog(null, "O valor total da prestação com juros é R$: "+ String.format("%.2f",valueTotal));

    }

    private static double simpleInterest(double interestRate, double valueAsset, int mouts) {
        double interestAmount;
        interestAmount = valueAsset * (interestRate / 100)* mouts;
        return interestAmount;
    }
}

File: Sem1/ProjetoIntegrador/exerciciosPyton/L01 - pg25_26Exercicio7/L01E.py
#e)Efetuar o cálculo e a apresentação do valor de uma prestação em atraso, 
# utilizando a fórmula PRESTACAO ← VALOR + (VALOR * TAXA/100) * TEMPO).

def calculo_prestacao(valor, taxa):
    total = valor + (valor * (taxa/100))
    return total
    
def verificacao_valor(valor):
    if valor <= 0:
        valor = False
    else:
        valor = True
    return valor

validation = False

while validation:
    try:
        valor = float(input("\nDigite o total do veiculo: ").replace(",","."))
    except:
        validation = verificacao_valor(valor)
        print("valor invalido, digite novamente: ")

validation = False
while validation ==False:
    taxa = float(input("\ndigite o valor da taxa em porcentagem (%): ").replace(",","."))
    validation = verificacao_valor(taxa)
    if validation == False:
        print("valor invalido, digite novamente: ")

prestacao = (calculo_prestacao(valor, taxa))

print(f"\nOvalor da prestação atualizada é: R${prestacao:.2f}")

